stock_putaway_product/models/product_putaway.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+# © 2016 Jos De Graeve - Apertoso N.V. <Jos.DeGraeve@apertoso.be>
+# © 2016 Carlos Dauden - Tecnativa <carlos.dauden@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import models, fields, api, _
+
+
+class ProductPutawayStrategy(models.Model):
+    _inherit = 'product.putaway'
+
+    @api.model
+    def _get_putaway_options(self):
+        ret = super(ProductPutawayStrategy, self)._get_putaway_options()
+        return ret + [('per_product', 'Fixed per product location')]
+
+    product_location_ids = fields.One2many(
+        comodel_name='stock.product.putaway.strategy',
+        inverse_name='putaway_id',
+        string='Fixed per product location',
+        copy=True)
+    method = fields.Selection(selection=_get_putaway_options)
+
+    @api.model
+    def putaway_apply(self, putaway_strategy, product):
+        if putaway_strategy.method == 'per_product':
+            offset = self.env.context.get('starting_strategy', 0)
+            strategy = putaway_strategy.product_location_ids.filtered(
+                lambda x: (x.product_product_id == product or
+                           (not x.product_product_id and
+                            x.product_tmpl_id == product.product_tmpl_id))
+            )[offset:offset+1]
```

Comment:
using `filtered` here seems very convoluted to me, especially since for some weird reason `offset` comes from the context. 

Using `[offset:offset+1]` over `[offset]` only makes sense if the offset may be greater than the number of filtered results, in which case `strategy` will be an empty resultset and the method returns `None`. 

Could you add some documentation on the behavior of this method?